app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+        $postDescription = $request->get('post_description', '');
+        $checksum = $request->get('checksum', 0);
+        $word_count = (int)$request->get('word_count', 0);
+
+        $autoSaveInterval = 4;
+
+        if($postId == 0 && empty($postContent) && empty($postTitle) && empty($postDescription)) {
+            return ['id' => 0];
+        }
+
+        if($request->session()->get('autoSave'.$postId) === null) {
+            $request->session()->set('autoSave'.$postId, [
+                'count' => 0,
+                'currentContent' => null
+            ]);
+        }
+
+        $autoSaver = $request->session()->get('autoSave'.$postId);
+
+        $calculatedChecksum = Post::calculateChecksum($postContent);
+
+        /** @var Post $post */
+        $post = Post::find($postId);
+
+        if(empty($post)) {
+            return ['error' => [
+                'message' => 'Post not found',
+                'code' => 404
+            ]];
+        }
+
+        $this->activityLog->setPostId($post->id);
+
+        if(!$post->canEditBy($this->currentUser)) {
+            $this->activityLog->addMeta('AutoSaveFalse', 'permission denied');
+            return ['error' => [
+                'message' => 'Post not found',
+                'code' => 404
+            ]];
+        }
+
+        // Lock detect
+        if($post->isLock($this->currentSession)) {
+            $this->activityLog->addMeta('AutoSaveFalse', 'expired session');
+            return [
+                'id' => $post->id,
+                'status' => 'locked'
+            ];
+        }
+
+        $post->content = $postContent;
+        $post->title = $postTitle;
+        $post->description = $postDescription;
+        $post->word_count = $word_count;
+
+        if(empty($checksum) && empty($postContent)) {
+            PostHistory::saveFromSession($request->session(), $post);
+            $this->activityLog->addMeta('AutoSaveFalse', 'post content is empty');
+            return [
+                'error' => [
+                    'message' => 'Post content is empty',
+                    'code' => 900
+                ]
+            ];
+        }
+
+        if(Post::calculateChecksum($postContent) != $checksum) {
+            PostHistory::saveFromSession($request->session(), $post);
+            $this->activityLog->addMeta('AutoSaveFalse', 'checksum failed');
+            return [
+                'error' => [
+                    'message' => 'Save failed',
+                    'trace' => 'Calculated checksum : '.$calculatedChecksum,
+                    'code' => 900
+                ]
+            ];
+        }
+
+        $autoSaver['currentContent'] = $post->content;
+        $autoSaver['count'] = (int)$autoSaver['count'] + 1;
+
+        $request->session()->set('autoSave'.$postId, $autoSaver);
+
+        $updatedTime = Carbon::now()->toDateTimeString();
+
+        //$history = null;
+
+        //if($autoSaver['count'] % PostHistory::AUTOSAVE_INTERVAL == 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS125)